mono/mini/mini.c

```diff
@@ -4272,3 +4272,20 @@ mini_class_is_system_array (MonoClass *klass)
 {
 	return m_class_get_parent (klass) == mono_defaults.array_class;
 }
+
+int mono_target_pagesize (void)
```

Comment:
Formating, we use do:
```
int
mono_target_pagesize (void)
{
...
```
This function is called from a signal so it must be audited + documented to be signal safe.

This means we need to audit the implementation of mono_pagesize to ensure they are signal safe as well.

Reply:
good point regarding signal safety. on some systems `sysconf (3)` is involved:
> POSIX.1-2001 and POSIX.1-2004 required the functions fpathconf(3), pathconf(3), and sysconf(3) to be async-signal-safe, but this requirement was removed in POSIX.1-2008.

we could add some initialisation logic, but we need to be super sure that this happens before a `SIGSEGV` can happen (sounds fragile?). I'm tempted to hardcode it to 4k for everything, it would simplify things a lot. What do you think?

Reply:
Let's just always use 4k as it's likely the smallest page size we'll ever encounter.
It's simpler and closes JIT/AOT behavior.

Reply:
sounds good 👍 I've updated the function including comment why the value is hard-coded.